substrate/client/transaction-pool/src/graph/pool.rs

```diff
@@ -320,18 +322,24 @@ impl<B: ChainApi, L: EventHandler<B>> Pool<B, L> {
 		// Option<Vec<Tag>>)`)
 		let all = extrinsics.iter().zip(in_pool_tags.into_iter());
 		let mut validated_counter: usize = 0;
-
 		let mut future_tags = Vec::new();
 		let now = Instant::now();
 		for (extrinsic, in_pool_tags) in all {
 			match in_pool_tags {
 				// reuse the tags for extrinsics that were found in the pool
 				Some(tags) => future_tags.extend(tags),
-				// if it's not found in the pool query the runtime at parent block
-				// to get validity info and tags that the extrinsic provides.
 				None => {
-					// Avoid validating block txs if the pool is empty
-					if !self.validated_pool.status().is_empty() {
+					// if it's not found in the pool, check if the extrinsic `provides`
```

Comment:
We now have two maps `(in_pool_hashes,in_pool_tags)` and `known_provides_tags`.

Both of them could be merged into a single map, by doing that you will avoid nested `if`.

Reply:
yep, I think it is the same suggestion as here, right? https://github.com/paritytech/polkadot-sdk/pull/7980#discussion_r2009846603

Reply:
resolved here: [6ac2eec](https://github.com/paritytech/polkadot-sdk/pull/7980/commits/6ac2eeccce2de1e5148b3429926b67b28ec1f301)